src/hooks/BalanceSummary.ts

```diff
@@ -63,48 +94,60 @@ const useBalanceSummary = (currentWalletInfo: WalletInfo | null): WalletBalanceS
         return acc
       }, {} as { [key: string]: Utxos })
 
+      const totalCalculatedByAccount = Object.fromEntries(
+        Object.entries(utxosByAccount).map(([account, utxos]) => {
+          return [account, utxos.reduce((acc, utxo) => acc + utxo.value, 0)]
+        })
+      )
       const frozenOrLockedCalculatedByAccount = Object.fromEntries(
         Object.entries(utxosByAccount).map(([account, utxos]) => {
-          return [account, calculateFrozenOrLockedBalance(utxos)]
+          return [account, calculateFrozenOrLockedBalance(utxos, refTime)]
         })
       )
 
-      const accountsBalanceSummary = accounts.map(({ account, account_balance, available_balance }) => {
-        const accountBalanceSummary: BalanceSummary = {
-          totalBalance: account_balance,
-          availableBalance: available_balance,
-        }
-
-        const accountFrozenOrLockedCalculated = frozenOrLockedCalculatedByAccount[account] || 0
-        const accountAvailableCalculatedInSats =
-          btcToSats(accountBalanceSummary.totalBalance!) - accountFrozenOrLockedCalculated
-        return {
-          ...accountBalanceSummary,
-          calculatedAvailableBalanceInSats: accountAvailableCalculatedInSats,
-          calculatedFrozenOrLockedBalanceInSats: accountFrozenOrLockedCalculated,
-          accountIndex: parseInt(account, 10),
-        } as AccountBalanceSummary
-      })
-
-      const walletFrozenOrLockedCalculated = Object.values(frozenOrLockedCalculatedByAccount).reduce(
+      const accountsBalanceSummary = accounts
+        .map(({ account, account_balance, available_balance }) => {
+          const accountBalanceSummary: BalanceSummary = {
+            totalBalance: account_balance,
+            availableBalance: available_balance,
+          }
+          const accountTotalCalculated: AmountSats = totalCalculatedByAccount[account] || 0
+          const accountFrozenOrLockedCalculated: AmountSats = frozenOrLockedCalculatedByAccount[account] || 0
+          const accountAvailableCalculated: AmountSats = accountTotalCalculated - accountFrozenOrLockedCalculated
+          return {
+            ...accountBalanceSummary,
+            calculatedTotalBalanceInSats: accountTotalCalculated,
+            calculatedFrozenOrLockedBalanceInSats: accountFrozenOrLockedCalculated,
+            calculatedAvailableBalanceInSats: accountAvailableCalculated,
+            accountIndex: parseInt(account, 10),
```

Comment:
NIT: Any reason for the explicit `10` in all the `parseInt` calls? 2nd param could be removed, 10 is default afaik.

Reply:
Over the years, I have made a habit of always indicating the radix to get reliable results.

> Be careful—this does not default to 10!
See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt

JavaScript is strange.

Edit: In this case, it would probably be fine, as we should not expect "malformed" inputs from JM.. nonetheless.. I guess I'll sleep better with the radix included. Is that okay for you?

Reply:
> JavaScript is strange.

Ain't that the truth! 😅